ksql-engine/src/test/resources/query-validation-tests/key-field.json

```diff
@@ -200,14 +293,93 @@
       ],
       "post": {
         "issues": [
-          "key field of output has INCORRECT name: should be 'ALIASED'."
-          ],
+          "key field of output has INCORRECT name: should be 'ALIASED' - (see Issue #4 above)"
+        ],
         "sources": [
           {"name": "INPUT", "type": "stream", "keyField": {"name": "FOO", "schema": {"type": "INT"}}},
           {"name": "OUTPUT", "type": "stream", "keyField": {"name": "FOO", "schema": {"type": "INT"}}}
         ]
       }
     },
+    {
+      "name": "stream | initially set | no key change | key in value | aliasing | legacy downstream query",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT foo as aliased, bar FROM INPUT;",
+        "CREATE STREAM DOWNSTREAM AS SELECT Aliased, bar FROM OUTPUT PARTITION BY Aliased;"
+      ],
+      "inputs": [
+        {"topic": "OUTPUT", "key": "1", "value": {"ALIASED":1, "BAR": 2}}
+      ],
+      "outputs": [
+        {"topic": "DOWNSTREAM", "key": "1", "value": {"ALIASED":1, "BAR": 2}}
+      ],
+      "post": {
+        "sources": [
+         {"name": "DOWNSTREAM", "type": "stream", "keyField": {"name": "ALIASED", "schema": {"type": "INT"}}}
+        ]
+      }
+    },
+    {
+      "name": "stream | initially set | no key change | key in value | aliasing + duplicate",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT foo as aliased, bar as foo FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "value": {"foo": 1, "bar": 2}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": {"ALIASED":1, "FOO": 2}}
+      ],
+      "post": {
+        "issues": [
+          "key field of output has INCORRECT name: should be 'ALIASED' - (see Issue #4 above)"
+        ],
+        "sources": [
+          {"name": "INPUT", "type": "stream", "keyField": {"name": "FOO", "schema": {"type": "INT"}}},
+          {"name": "OUTPUT", "type": "stream", "keyField": {"name": "FOO", "schema": {"type": "INT"}}}
+        ]
+      }
+    },
+    {
+      "name": "stream | initially set | no key change | key in value | aliasing + duplicate | legacy downstream query | no key change",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT foo as aliased, bar as foo FROM INPUT;",
+        "CREATE STREAM DOWNSTREAM AS SELECT aliased, foo FROM OUTPUT PARTITION BY aliased;"
+      ],
+      "inputs": [
+        {"topic": "OUTPUT", "key": "1" ,"value": {"ALIASED":1, "FOO": 2}}
```

Comment:
Same question as the ones below, why not generate the data into `INPUT`?

Reply:
QTT only starts the last topology / statement, so you need to generate input for that topology.